mem_concat.go

```diff
@@ -240,6 +241,10 @@ func tarGroup(ctx context.Context, client *s3.Client, objectList []*S3Obj) ([]by
 			AccessTime: *o.LastModified,
 			Format:     tarFormat,
 		}
+		if preservePOSIXMetadata {
+			head := fetchS3ObjectHead(ctx, client, o)
```

Comment:
The `GetObject` API call returns a Metadata dictionary as well. It might be worth modifying the `downloadS3Data` function so it also returns the header data that you need. Have you checked to see if it contains what is needed to presesrve the POSIXMetadata?

Reply:
Good catch! I have implemented this.